packages/grid/_modules_/grid/components/menu/columnMenu/HideGridColMenuItem.tsx

```diff
@@ -35,7 +38,7 @@ const HideGridColMenuItem = (props: GridFilterItemProps) => {
   }
 
   return (
-    <MenuItem onClick={toggleColumn}>
+    <MenuItem onClick={toggleColumn} disabled={visibleColumnsLength === 1}>
```

Comment:
Only this check will not be enough. Some columns have the column menu disabled with `disableColumnMenu`, like the checkbox selection. The user may ended up with only one column and this column doesn't have the menu. I think the correct assertion is to check if there's only one column with the column menu not disabled.

<img width="150" src="https://user-images.githubusercontent.com/42154031/149441949-1e6d108c-590c-4b69-8935-b03ba700afe9.png" alt="image" style="max-width: 100%;">

Reply:
Right!
Added a test for this scenario as well